src/constant_value.rs

```diff
@@ -52,15 +56,21 @@ impl ConstantValue {
 
 /// Keeps track of MIR constants that have already been mapped onto ConstantValue instances.
 pub struct ConstantValueCache {
-    cache: HashMap<DefId, ConstantValue>,
+    function_cache: HashMap<DefId, ConstantValue>,
```

Comment:
Just to be pedantic, these aren't actually caches, right? They're just lookup tables?

Reply:
It depends on what you mean by lookup table. I do think of them as caches myself. I guess you could also call them intern tables. The basic idea is that these ConstantValues are created on demand and then cached so that future calls just look them up.